socorro/external/fs/crashstorage.py

```diff
@@ -663,68 +659,115 @@ class TarFileCrashStore(CrashStorageBase):
+        self.config.logger.debug(
+            'TarFileCrashStore saved - %s to %s',
+            crash_id,
+            self.config.tarball_name
         )
-        file_handle = self.gzip_module.open(file_name, 'w', 9)
-        try:
-            file_handle.write(processed_crash_as_string)
-        finally:
-            file_handle.close()
-        self._save_to_tarfile(
-            file_name,
-            os.path.join(
-                crash_id[:2],
-                crash_id[2:4],
-                crash_id + '.jsonz'
-            )
+
+
+#==============================================================================
+class TarFileSequentialReadingCrashStore(CrashStorageBase):
+    required_config = Namespace()
+    required_config.add_option(
+        name='tarball_name',
+        doc='pathname to a the target tarfile',
+        default='fred.tar'
+    )
+    required_config.add_option(
+        name='tarfile_module',
+        doc='a module that supplies the tarfile interface',
+        default='tarfile',
+        from_string_converter=class_converter
+    )
+    required_config.add_option(
+        name='gzip_module',
+        doc='a module that supplies the gzip interface',
+        default='gzip',
+        from_string_converter=class_converter
+    )
+
+    #------------------------------------------------------------------------------
+    @staticmethod
+    def stringify_datetimes(obj):
+        if isinstance(obj, datetime.datetime):
+            return obj.strftime("%Y-%m-%dT%H:%M:%S.%f")
+        return json.JSONEncoder.default(self, obj)
+
+    #------------------------------------------------------------------------------
+    def _create_tarfile(self):
+        """subclasses that have a different way of openning or creating
+        the tar file pointer can override this method.  Useful for creating
+        text buffer tarfiles or using temporary files"""
+        return self.tarfile_module.open(self.config.tarball_name, 'r')
+
+    #------------------------------------------------------------------------------
+    def __init__(self, config, quit_check_callback=None):
+        super(TarFileSequentialReadingCrashStore, self).__init__(
+            config,
+            quit_check_callback
+        )
+        self.tarfile_module = config.tarfile_module
+        self.gzip_module = config.gzip_module
+        self.tar_fp = self._create_tarfile()
+
+    #------------------------------------------------------------------------------
+    def close(self):
+        self.tar_fp.close()
+
+    #------------------------------------------------------------------------------
+    def get_unredacted_processed(self, crash_id_ignored):
+        """we don't implement random access in this class, the next
+        one is all you get no matter what you ask for"""
+        a_tar_info_object = self.tar_fp.next()
+        if a_tar_info_object is None:
+            raise CrashIDNotFound(crash_id_ignored)
```

Comment:
This isn't very helpful. I think it should be `raise CrashIDNotFound('no next tar object available')`